src/rdkafka.h

```diff
@@ -3428,6 +3434,39 @@ rd_kafka_resp_err_t rd_kafka_seek (rd_kafka_topic_t *rkt,
                                    int timeout_ms);
 
 
+
+/**
+ * @brief Seek consumer for partitions in \p partitions to the per-partition
+ *        offset in the \c .offset field of \p partitions.
+ *
+ * The offset may be either absolute (>= 0) or a logical offset.
+ *
+ * If \p timeout_ms is not 0 the call will wait this long for the
+ * seeks to be performed. If the timeout is reached the internal state
+ * will be unknown for the remaining partitions to seek and this function
+ * will return an error with the error code set to
+ * `RD_KAFKA_RESP_ERR__TIMED_OUT`.
```

Comment:
i think it would be good to add a short summary here about what happens when you call seek. enough so that the user can understand why it might block and that it currently flushes the message cache.

also in `rd_kafka_seek`